src/model_signing/_signing/sign_certificate.py

```diff
@@ -141,11 +142,22 @@ def __init__(
     def _verify_bundle(self, bundle: bundle_pb.Bundle) -> tuple[str, bytes]:
         public_key = self._verify_certificates(bundle.verification_material)
         envelope = bundle.dsse_envelope
-        public_key.verify(
-            envelope.signatures[0].sig,
-            sigstore_pb.pae(envelope.payload),
-            ec.ECDSA(ec_key.get_ec_key_hash(public_key)),
-        )
+        try:
+            public_key.verify(
+                envelope.signatures[0].sig,
+                sigstore_pb.pae(envelope.payload),
+                ec.ECDSA(ec_key.get_ec_key_hash(public_key)),
+            )
+        except exceptions.InvalidSignature:
+            # Compatibility layer with pre 1.0 release
+            # Here, we patch over a bug in `pae` which mixed unicode `str` and
+            # `bytes`. As a result, additional escape characters were added to
+            # the material that got signed over.
```

Comment:
do we want to retry all signature failures? Or just ones that declare the compatibility type?`_PREDICATE_TYPE_COMPAT`

Reply:
We would need to do an inversion of control to check the predicate type. At this layer the predicate is just a sequence of bytes, we would need to parse it, check the predicate type and then verify that again. And then, we do the same checks again at the proper place.

The exceptions raised from the `except` block are the same as those raised from `try`, so I'd assume this would be safe.

Reply:
That was my other question, about the exception chaining. It seems like there would just be more context from both exceptions if both got raised.

Reply:
It's the same exception though. The only additional information is that this would be from a different line number.